### ISSUE TITLE
Clean up obsolete modules

### ISSUE THREAD

**Issue by wei2912** [apertium/phenny#436]
Look through the list of modules and determine which ones can be deprecated. Modules to be deprecated should either be 1. very rarely used, 2. NSFW, 3. unrepairable or 4. obsolete. Deprecated modules should be shifted into "modules/archive".

**Comment by nu11us** [comment]
Yeah, I agree with that. @wei2912 mind if I just delete the obsolete files?

**Comment by wei2912** [comment]
@Androbin fair point, a delete of the modules should be more appropriate.

**Comment by sushain97** [comment]
Personally, I would prefer nsfw modules to just be default disabled under an nsfw flag in the config.

**Comment by nu11us** [comment]
They do appear publicly on this repository though and to anyone working on the IRC bot. 

Having them, especially `rule34.py`, out in the open to a competition of 13-17 year olds doesn't sound appropriate, even if they are disabled.

**Comment by shardulc** [comment]
I'm in favor of outright deleting the NSFW modules and perhaps even other modules that never get used, like `mylife.py`.

**Comment by nu11us** [comment]
That's in the pull request as of now. I've also uploaded a script in my task that gives the usage of all the scripts from the logs since around 2010.

**Comment by nu11us** [comment]
*Usage as of now with a few false positives*
https://pastebin.com/eKDm5wuC

**Comment by nu11us** [comment]
Output: https://pastebin.com/15sjKs93

**Comment by shardulc** [comment]
Going along with what @wei2912 said, what about removing all modules used less than X times? (He suggests X=50, I think X=20 would be better, thoughts?) Caveat: there are some modules that aren't actually 'called', but rather respond to text in the channel. For example, whenever someone says "particle", begiak responds. These should not be deleted.

**Comment by nu11us** [comment]
I considered that, but the some of the unused modules still have some utility or charm. Looking through the output pastebin, I think it would have to be around n=5 as the majority of all the commands are under 20.

**Comment by shardulc** [comment]
The ones that have "charm" can be picked out by hand but I think it's fine to remove the majority of them. @jonorthwash may have a different opinion on this, though. Meanwhile could you compile a list of the ones below 20 and see whether any of those are actually dependencies for others, have a 'reaction' like particles, need to be kept for some other reason, etc.?

**Comment by nu11us** [comment]
| Command               	| Uses 	| Deleted? 	|   	|   	|
|-----------------------	|------	|----------	|---	|---	|
| autojoin              	| 0    	| No       	|   	|   	|
| checkMessages         	| 0    	| No       	|   	|   	|
| f_weather             	| 0    	| No       	|   	|   	|
| foodvote              	| 0    	| Yes      	|   	|   	|
| generate              	| 0    	| No       	|   	|   	|
| harglebargleP         	| 0    	| No       	|   	|   	|
| mailing list reporter 	| 0    	| No       	|   	|   	|
| nightnight            	| 0    	| No       	|   	|   	|
| perword               	| 0    	| No       	|   	|   	|
| pointing              	| 0    	| No       	|   	|   	|
| retrieve_commit       	| 0    	| No       	|   	|   	|
| uderp                 	| 0    	| No       	|   	|   	|
| unsupervised          	| 0    	| No       	|   	|   	|
| vtluug                	| 0    	| Yes      	|   	|   	|
| bytes                 	| 1    	| No       	|   	|   	|
| food                  	| 1    	| Yes  |   	|   	|
| get_recent_commit     	| 1    	| No       	|   	|   	|
| identlang             	| 1    	| No       	|   	|   	|
| information           	| 1    	| No       	|   	|   	|
| particles             	| 1    	| No       	|   	|   	|
| pester stop           	| 1    	| No       	|   	|   	|
| quit                  	| 1    	| No       	|   	|   	|
| upgrade               	| 1    	| No       	|   	|   	|
| wadsworth             	| 1    	| No       	|   	|   	|
| calccoverage          	| 2    	| No       	|   	|   	|
| ethnologue            	| 2    	| No       	|   	|   	|
| head                  	| 2    	| No       	|   	|   	|
| linx                  	| 2    	| No       	|   	|   	|
| apystats              	| 3    	| No       	|   	|   	|
| bargle                	| 3    	| No       	|   	|   	|
| hargle                	| 3    	| No       	|   	|   	|
| nsfw                  	| 3    	| No       	|   	|   	|
| part                  	| 3    	| No       	|   	|   	|
| suggest               	| 3    	| No       	|   	|   	|
| choose                	| 4    	| No       	|   	|   	|
| udmurt                	| 4    	| No       	|   	|   	|
| wuvt                  	| 4    	| Yes      	|   	|   	|
| bothug                	| 5    	| No       	|   	|   	|
| hs                    	| 5    	| No       	|   	|   	|
| fcc                   	| 6    	| Yes      	|   	|   	|
| join                  	| 7    	| No       	|   	|   	|
| stache                	| 7    	| No       	|   	|   	|
| analyse               	| 8    	| No       	|   	|   	|
| harglebargle          	| 10   	| No       	|   	|   	|
| stats                 	| 12   	| No       	|   	|   	|
| ety                   	| 13   	| No       	|   	|   	|
| pesters               	| 13   	| No       	|   	|   	|
| slogan                	| 13   	| No       	|   	|   	|
| botslap               	| 14   	| No       	|   	|   	|
| following             	| 14   	| No       	|   	|   	|
| pester                	| 14   	| No       	|   	|   	|
| commit                	| 16   	| No       	|   	|   	|
| imdb                  	| 16   	| No       	|   	|   	|
| unfollow              	| 17   	| No       	|   	|   	|

**Comment by jonorthwash** [comment]
I'd keep `fcc`.  It's off-topic for #apertium, but it's a useful module and it still works.

**Comment by nu11us** [comment]
Will add it back momentarily. Any others?

**Comment by Androbin** [comment]
I would argue that `vtluugwiki.py` (and the other wikis) was reduced to so little extra code that it's near zero maintenance.
Instead of removing it, we might as well have a combined wiki module which enumerates (with very little overhead) the set of wiki URLs (+names).
Then again, only the general wiki code that's already used in/by/for `apertium_wiki.py` will have to be maintained.
I suggest keeping them for now and creating another ticket (potential GCI task) for implementing such changes.

**Comment by nu11us** [comment]
Yeah, I was on the side of keeping them at one point due to potential, but they still remain unused. Perhaps, we could remove them for now until a combined wiki module is implemented? That would allow me to finish this task and have another GCI task for that.

**Comment by wei2912** [comment]
Consider the remaining modules here:

| *Module* | *Reason for Removal* |
| archwiki.py | We don't even use ArchWiki |
| eleda.py | Doesn't seem like we've ever use it |
| imdb.py | kinda useful but not sure if it's working |
| lastfm.py | do we really want to know what users are listening to |

**Comment by nu11us** [comment]
1. ArchWiki has been used quite a lot, at least according to the logs.
2. Eleda was custom-written, so I didn't feel like deleting it.
3. Imdb works
4. I suggested that we deleted lastfm.

**Comment by sushain97** [comment]
I'm guessing ArchWiki usages are probably Apertium Wiki usages.

**Comment by nu11us** [comment]
That explains it.

**Comment by nu11us** [comment]
Just removed ArchWiki.

**Comment by wei2912** [comment]
Closing as the PR has been merged.

**Comment by jonorthwash** [comment]
Wait, were the imdb and weather modules deleted too?

**Comment by jonorthwash** [comment]
> I suggest keeping them for now and creating another ticket (potential GCI task) for implementing such changes.

+1, was this ticket opened?

**Comment by nu11us** [comment]
Yeah, imdb was broken and weather was rarely used.

**Comment by jonorthwash** [comment]
> imdb was broken

Ah, okay.

> and weather was rarely used.

That's a good one to keep around though...  And it could be more useful to another community.  Is there a way to bring it back?

**Comment by nu11us** [comment]
Of course; I can just add another pull request or someone can just re-add the module to the folder.

**Comment by jonorthwash** [comment]
@nu11us, could you just add another PR?

**Comment by wei2912** [comment]
Given that upstream has been archived, we may want to relook through all the modules and remove those that we won't plan to actively develop.

**Comment by scoopgracie** [comment]
So, should we really keep the ones listed as "Charm"? Like, is having .bot*, harglebargle, noexceptions, etc. really useful?
